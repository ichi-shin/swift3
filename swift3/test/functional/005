#!/bin/bash

# DELETE Bucket

. ./common

S3USER=tester

_s3_put /bucket -H 'x-amz-acl: public-read'

upload_id=$(_s3_post /bucket/object?uploads | \
    _xpath '/InitiateMultipartUploadResult/UploadId/text()')
_s3_put /bucket/object?uploadId=${upload_id}\&partNumber=1 -T /dev/null

echo '<?xml version="1.0" encoding="UTF-8"?>
<BucketLoggingStatus xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <LoggingEnabled>
    <TargetBucket>bucket2</TargetBucket>
    <TargetPrefix/>
  </LoggingEnabled>
</BucketLoggingStatus>' > $tmp.logging.xml
_s3_put /bucket2 -H 'x-amz-acl: log-delivery-write'
_s3_put /bucket?logging -T $tmp.logging.xml

echo '<LifecycleConfiguration>
<Rule>
   <ID>Archive and then delete rule</ID>
   <Prefix>prefix_</Prefix>
   <Status>Enabled</Status>
   <Expiration><Days>0</Days></Expiration>
</Rule>
</LifecycleConfiguration>' > $tmp.lifecycle.xml
MD5=$(_md5 $tmp.lifecycle.xml) _s3_put /bucket?lifecycle -T $tmp.lifecycle.xml
_s3_delete /bucket?lifecycle

echo '<VersioningConfiguration xmlns="http://s3.amazonaws.com/doc/2006-03-01/"> 
<Status>Enabled</Status>
</VersioningConfiguration>' > $tmp.versioning.xml
_s3_put /bucket?versioning -T $tmp.versioning.xml

version_id[0]=$(_s3_put /bucket/object -T /dev/null \
    -H 'x-amz-acl: public-read' -D - | _hq 'x-amz-version-id')
version_id[1]=$(_s3_put /bucket/object -T $tmp.versioning.xml \
    -H 'x-amz-acl: public-read-write' -D - | _hq 'x-amz-version-id')
_s3_delete /bucket/object?versionId=${version_id[0]} -D - | _filter_curl
_s3_delete /bucket/object?versionId=${version_id[1]} -D - | _filter_curl

_s3_delete /bucket -D - | _filter_curl

_sw_get /bucket | _filter_curl xml
_sw_get /bucket+versions | _filter_curl
_sw_get /bucket+segments | _filter_curl | sed -e "s/${upload_id}/UPLOAD_ID/g"
