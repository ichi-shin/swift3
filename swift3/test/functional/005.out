QA output created by 005
> s3curl --id tester -- -X PUT -H x-amz-acl: public-read http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X POST http://SWIFT_HOST/bucket/object?uploads... 200
> s3curl --id tester -- -X PUT -T /dev/null http://SWIFT_HOST/bucket/object?uploadId=UPLOAD_ID&partNumber=1... 200
> s3curl --id tester -- -X PUT -H x-amz-acl: log-delivery-write http://SWIFT_HOST/bucket2... 200
> s3curl --id tester -- -X PUT -T /TMP.logging.xml http://SWIFT_HOST/bucket?logging... 200
> s3curl --id tester --contentMd5 U2caJjlmWSTwONBlU/U/dA== -- -X PUT -T /TMP.lifecycle.xml http://SWIFT_HOST/bucket?lifecycle... 200
> s3curl --id tester -- -X DELETE http://SWIFT_HOST/bucket?lifecycle... 204
> s3curl --id tester -- -X PUT -T /TMP.versioning.xml http://SWIFT_HOST/bucket?versioning... 200
> s3curl --id tester -- -X PUT -T /dev/null -H x-amz-acl: public-read -D - http://SWIFT_HOST/bucket/object... 200
> s3curl --id tester -- -X PUT -T /TMP.versioning.xml -H x-amz-acl: public-read-write -D - http://SWIFT_HOST/bucket/object... 200
> s3curl --id tester -- -X DELETE -D - http://SWIFT_HOST/bucket/object?versionId=VERSION_ID... 204
HTTP/1.1 204 No Content
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

> s3curl --id tester -- -X DELETE -D - http://SWIFT_HOST/bucket/object?versionId=VERSION_ID... 204
HTTP/1.1 204 No Content
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

> s3curl --id tester -- -X DELETE -D - http://SWIFT_HOST/bucket... 204
HTTP/1.1 204 No Content
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

> curl -X GET http://SWIFT_HOST/v1/TENANT/bucket... 404
<?xml version='1.0' encoding='UTF-8'?>
<html>
  <h1>Not Found</h1>
  <p>The resource could not be found.</p>
</html>
> curl -X GET http://SWIFT_HOST/v1/TENANT/bucket+versions... 204
> curl -X GET http://SWIFT_HOST/v1/TENANT/bucket+segments... 200
object/UPLOAD_ID
object/UPLOAD_ID/1
